lib/dialects/abstract/query-generator.js

```diff
@@ -181,11 +182,12 @@ class QueryGenerator {
     let onDuplicateKeyUpdate = '';
 
     if (this._dialect.supports.inserts.updateOnDuplicate && options.updateOnDuplicate) {
-      if (this._dialect.supports.inserts.updateOnDuplicate == ' ON CONFLICT DO UPDATE SET') { // postgres / sqlite
+      if (this._dialect.supports.inserts.updateOnDuplicate === ' ON CONFLICT DO UPDATE SET') { // postgres / sqlite
         // If no conflict target columns were specified, use the primary key names from options.upsertKeys
         const conflictKeys = options.upsertKeys.map(attr => this.quoteIdentifier(attr));
         const updateKeys = options.updateOnDuplicate.map(attr => `${this.quoteIdentifier(attr)}=EXCLUDED.${this.quoteIdentifier(attr)}`);
-        onDuplicateKeyUpdate = ` ON CONFLICT (${conflictKeys.join(',')}) DO UPDATE SET ${updateKeys.join(',')}`;
+        const maybeWhere = this.whereQuery(conflictWhere);
+        onDuplicateKeyUpdate = ` ON CONFLICT (${conflictKeys.join(',')}) ${maybeWhere ? `${maybeWhere} ` : ''}DO UPDATE SET ${updateKeys.join(',')}`;
```

Comment:
Just a small thing, but `${maybeWhere ? `${maybeWhere} ` : ''}` in L190 - L298 can be optimized to `${maybeWhere || ''}`.

Additionally, it might make even more sense to abstract the above lines into a function, since they share the same functionality as L295 - L298 (but this might already be out of scope of this PR)

Reply:
Opted not to do this since I couldn't think of a good name for an abstract function + it doesn't make sense (imo) to make a function specifically for this case (and since this only used twice + isn't much logic).